src/openms/source/KERNEL/ConsensusMap.cpp

```diff
@@ -559,11 +561,31 @@ namespace OpenMS
     Size i(0);
     for (auto const & p : s)
     {
+      if (!p.hasSuffix("mzML"))
+      {
+        OPENMS_LOG_WARN << "To ensure tracability of results please prefer mzML files as primary MS run." << std::endl
+                        << "Filename: '" << p << "'" << std::endl;                          
+      }
+
       column_description_[i].filename = p;
       ++i;
     }
   }
 
+  void ConsensusMap::setPrimaryMSRunPath(const StringList& s, MSExperiment & e)
+  {
+    StringList ms_path;
+    e.getPrimaryMSRunPath(ms_path);
+    if (ms_path.size() == 1 && ms_path[0].hasSuffix("mzML") && File::exists(ms_path[0]))
```

Comment:
@timosachsenberg what if you run this on a different system, e.g. on a cluster where the original file path is invalid? I am not sure that `File::exists(ms_path[0])` is a always a good idea...

Reply:
The reasoning is that we carry over a file that exists on the file system. If not stored in the mzML, then from the input to the current tool execution.
We could in principle remove the check if the file exists but then we face the following problem:
If a user converted a raw file or mzML file at a different place/machine (which I think might be quite common) this filename is carried over to the end of the workflow and it is unlikely that we can open it e.g., in TOPPView.

Reply:
E.g., in many workflow systems, the input files are accessible on all nodes but the intermediate files might be on the nodes temporary folder. This scenario works with current implementation (it will set the proper path to the mzML in the input folder in the first tool) but fails if the check is ommited (and the mzML contains an inaccessible location e.g. from the instrument machine).

Reply:
The question is whether we care about keeping track of the file or whether we want to be able to open the file itself later on (seems like a much harder task). I agree that under most conditions it should still work, I just feel that checking whether the file exists on that system is not robust and will thus produce different results when run on test/development systems etc. Seems like a way to introduce hard to track / reproduce bugs

>  If a user converted a raw file or mzML file at a different place/machine (which I think might be quite common) this filename is carried over to the end of the workflow and it is unlikely that we can open it e.g., in TOPPView.

I assumed that is the point, to track where the file came from? So that I can track which input file produced which output file

> E.g., in many workflow systems, the input files are accessible on all nodes but the intermediate files might be on the nodes temporary folder. This scenario works with current implementation (it will set the proper path to the mzML in the input folder in the first tool) but fails if the check is ommited (and the mzML contains an inaccessible location e.g. from the instrument machine).

this will not work with docker where the file paths are not global for the system